<launch>
	<group ns="stereo">
		<node pkg="uvc_camera" type="uvc_camera_node" name="left">
		    <remap from="image_raw" to="left/image_raw" />
		    <remap from="camera_info" to="left/camera_info" />
		    <remap from="camera" to="left" />

			<param name="device" value="/dev/video1"/>
			<param name="camera_info_url" value="file://$(find carrt_goggles)/data/left.yaml"/>

			<param name="width" value="640"/>
			<param name="height" value="480"/>
			<param name="fps" value="60"/>
			<param name="frame_id" value="left"/>

			<param name="exposure_absolute" value="110"/>
		</node>

		<node pkg="uvc_camera" type="uvc_camera_node" name="right">
			<remap from="image_raw" to="right/image_raw" />
			<remap from="camera_info" to="right/camera_info" />
			<remap from="camera" to="right" />

			<param name="device" value="/dev/video2"/>
			<param name="camera_info_url" value="file://$(find carrt_goggles)/data/right.yaml"/>

			<param name="width" value="640"/>
			<param name="height" value="480"/>
			<param name="fps" value="60"/>
			<param name="frame_id" value="right"/>

			<param name="exposure_absolute" value="110"/>
		</node>

		<node pkg="carrt_goggles" type="fisheye_rect" name="fisheye_rect" />

		<node pkg="carrt_goggles" type="obstacle_detector" name="obstacle_detector">
			<remap from="obstacles/warning" to="/audio_buffer/obstacle" />
		</node>

		<node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
		<node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load fisheye_rect $(find carrt_goggles)/cfg/fisheye_cam.yaml" />
	</group>

	<group ns="audio_buffer">
		<node name="soundplay_node" pkg = "sound_play" type = "soundplay_node.py"/>
		<!--node name="say" pkg = "carrt_goggles" type = "say.py"/-->
		<node pkg= "carrt_goggles" type = "audio_buffer_node" name = "audio_buffer_node"/>
	</group>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find carrt_goggles)/launch/fisheye_cam.rviz" />
</launch>